File: DSA Practice/LeetCode/242. Valid Anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> m = new HashMap<Character, Integer>();

        if (s.length() != t.length()){
            return false;
        }

        for (char c : s.toCharArray()){
            if (m.containsKey(c)){
                m.put(c, m.get(c) + 1);
            } else {
                m.put(c, 1);
            }
        }

        for (char c : t.toCharArray()){
            if (m.containsKey(c)){
                if (m.get(c) > 0){
                    m.put(c, m.get(c) - 1);
                } else {
                    return false;
                }
            } else {
                return false;
            }
        }

        return true;
    }
}

File: DSA Practice/LeetCode/242. Valid Anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m = {}
        
        sn = len(s)
        tn = len(t)

        if sn != tn:
            return False

        for i in range(sn):

            if s[i] in m.keys():
                m[s[i]] += 1

            else :
                m[s[i]] = 1

        for i in range(tn):
            if t[i] in m.keys():

                if m[t[i]] > 0:
                    m[t[i]] -= 1

                else:
                    return False

            else:
                return False

        return sum(m.values()) == 0 